proposals/separate-standalone-driver/README.md

```diff
@@ -0,0 +1,190 @@
+# Standalone driver 
+
+## Summary
+In [Kubeflow Pipelines](https://www.kubeflow.org/docs/components/pipelines/interfaces/), each node in the pipeline graph is non-atomic and, at the Kubernetes level, consists of two components: a driver and an executor. Each of these runs in a separate Kubernetes pod. Additionally, every pipeline run spawns a root DAG driver pod.
+
+Here’s a simple diagram of the pods created during a KFP (Kubeflow Pipelines) run:
+
+*(User-defined pipeline with 2 tasks: Task 1 and Task 2)*
+
+![img.png](kfp-pods.png)
+
+This proposal explores approaches to replacing both the root DAG driver and all container drivers with a single standalone service, using Argo Workflows as the orchestration framework.
+
+
+## Motivation
+While using a separate pod for the executor makes sense - since it often handles heavy workloads and benefits from isolation and flexibility - the driver is a lightweight component. It typically performs just a few API calls: checking for cached results and creating an MLMD Execution.
+
+However, running the driver in a separate pod causes several issues:
+
+High overhead: Launching a Kubernetes pod merely to execute a few API calls introduces significant latency. Often, the pod scheduling and startup time outweighs the driver's actual processing time.
+
+Resource availability problems: There's no guarantee the Kubernetes cluster has sufficient resources to schedule the driver pod. If scheduling fails, the pipeline gets stuck. The UI currently doesn't show driver pod scheduling failures, which makes it hard to debug and understand what's going on.
+
+## Current state details
+
+Let's take a look at the copy of [hello_world.yaml](hello_world.yaml)  generated by the argo compiler tests.
+
+Execution Order:
+1. entrypoint
+   Type: DAG
+The root DAG represents the entire pipeline run.
+2. root-driver (template: system-dag-driver)
+   Type: Container
+   Purpose: Initializes the root DAG. Creates an MLMD execution for the DAG.
+Outputs:
+execution-id – ID of the DAG execution (created during root-driver execution)
```

Comment:
These line breaks are visible in the raw markdown but appear to be one line in the rendered markdown. Truly a nit, but mind making sure that they render properly?